CDP4IME/ViewModels/DataSourceExportViewModel.cs

```diff
@@ -274,24 +278,49 @@ private set
         /// </summary>
         private async void ExecuteOk()
         {
+            var fileInfo = new FileInfo(this.Path);
+
+            // check file exists and warn of override
+            if (fileInfo.Exists)
+            {
+                var overrideMessageBox = ThemedMessageBox.Show(title: "File Exists", text: "The selected output file already exists. Would you like to override?", messageBoxButtons: MessageBoxButton.YesNo, icon: MessageBoxImage.Question);
+
+                if (overrideMessageBox == MessageBoxResult.No)
+                {
+                    return;
+                }
+            }
+
+            // check directory exists
+            if (!Directory.Exists(fileInfo.DirectoryName))
+            {
+                this.ErrorMessage = "The output directory does not exist.";
+                return;
+            }
+
             this.IsBusy = true;
             this.LoadingMessage = "Exporting...";
 
             try
             {
                 var creds = new Credentials(this.selectedSession.Credentials.UserName, this.Password, new Uri(this.Path));
-                
-                // TODO: change this to allow (file) dal selection for export
-                var dal = this.dals.Single(x => x.Metadata.DalType == DalType.File);
+
+                var dal = this.dals.Single(x => x.Metadata == this.SelectedDal);
                 var dalInstance = (IDal)Activator.CreateInstance(dal.Value.GetType());
-                
+
                 var fileExportSession = dalInstance.CreateSession(creds);
 
-                // create write 
+                // create write
                 var operationContainers = new List<OperationContainer>();
 
                 // TODO: allow iteration setup selection by user
-                var openIterations = this.selectedSession.OpenIterations.Select(x => x.Key);
+                var openIterations = this.selectedSession.OpenIterations.Select(x => x.Key).ToList();
+
+                if (!openIterations.Any())
+                {
+                    throw new Exception("There must be at least one open Iteration in the selected Session.");
```

Comment:
i propose we make this an explicit exception (IterationNotFoundException) or something like that, so that when it is caught a relevant message can be popped to the user